Additonal improvements for the existing servlets

sundaramrajendran opened the pull request:
- All of our console pages html content loaded from templates instead of html code intermixed in the java code
- All of our html, js, css and image resources moved to new path under src/main/resources/com/paypal/selion
- Added SeLion Artwork logo to all console pages
- Updated GridForceRestartServlet as inject application/json into the template which jquery javascript uses to render the page

sundaramrajendran commented:
@mach6 / @sebady  Please review this PR

mach6 commented on server/src/main/java/com/paypal/selion/grid/servlets/LoginServlet.java:
```diff
@@ -23,6 +23,7 @@
 import javax.servlet.http.HttpServletResponse;
 import javax.servlet.http.HttpSession;
 
+import org.apache.commons.io.IOUtils;
```
This import is not used.

mach6 commented on server/src/main/java/com/paypal/selion/utils/ServletHelper.java:
```diff
@@ -37,6 +37,9 @@
  * 
  */
```
The javadoc talks about commonly repeated HTML code .. which is no longer the case. :)

mach6 commented on server/src/main/java/com/paypal/selion/grid/servlets/GridForceRestartDelegateServlet.java:
```diff
@@ -107,55 +114,26 @@ protected void process(HttpServletRequest request, HttpServletResponse response)
+            final String json = new GsonBuilder().serializeNulls().create().toJson(proxies);
+            final String jsonUtf8 = new String(json.getBytes(), "UTF-8");
+            String template = IOUtils.toString(this.getClass().getResourceAsStream(RESOURCE_PAGE_FILE), "UTF-8");
+            writer.write(String.format(template, jsonUtf8, GridForceRestartDelegateServlet.class.getSimpleName()));
```
I think we should create a `ServletHelper#respondAsHtmlUsingJsonAndTemplateWithHttpStatus(HttpServletResponse resp, Object toJson, String resourcePageTemplate, int statusCode, String ... additionalStrings)` to use here and in other places.

mach6 commented on server/src/main/java/com/paypal/selion/grid/servlets/GridAutoUpgradeDelegateServlet.java:
```diff
@@ -94,11 +95,9 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws S
-     *            - {@link HttpServletResponse} that represent the servlet response
+     *
+     * @param request  - {@link HttpServletRequest} that represent the servlet request
+     * @param response - {@link HttpServletResponse} that represent the servlet response
```
please remove the `-`'s.. Also this file exceeds the 120 char margin

mach6 commented on server/src/main/resources/com/paypal/selion/html/listAllNodes.html:
```diff
@@ -29,7 +29,8 @@
 
 <body id="main_body">
-<img id="top" src="/grid/resources/form/top.png" alt="">
+<img src='/grid/resources/com/paypal/selion/image/selion-horiz-notag.png' alt='SeLion Logo, horizontal' border='0'>
```
Let's be consistent. The rest of this file uses `"` around attribute values.

mach6 commented on server/src/main/resources/com/paypal/selion/html/listAllNodes.html:
```diff
@@ -29,7 +29,8 @@
 <body id="main_body">
-<img id="top" src="/grid/resources/form/top.png" alt="">
+<img src='/grid/resources/com/paypal/selion/image/selion-horiz-notag.png' alt='SeLion Logo, horizontal' border='0'>
+<img id="top" src="/grid/resources/com/paypal/selion/image/top.png" alt="">
```
Let's provide the `alt` text

mach6 commented on server/src/main/resources/com/paypal/selion/html/updateSauceConfigPage.html:
```diff
@@ -0,0 +1,85 @@
+      </li>
+      <li class='buttons'><input type='hidden' name='form_id' value='login'/> <input id='saveForm'
+                                                                                     class='button_text' type='submit'
+                                                                                     name='submit' value='Submit'/></li>
```
Can we fix these lines so they do not line break at the attributes like this?

mach6 commented:
Will you please make all of the .js and .html files use a 2 space indent like we did in SeLion client's code?

mach6 commented:
@sundaramrajendran minor comments aside, this looks good to me. thanks!

sundaramrajendran commented:
@mach6 I have addressed all the review comments

mach6 commented on server/src/main/java/com/paypal/selion/grid/servlets/GridForceRestartDelegateServlet.java:
```diff
@@ -15,23 +15,23 @@
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
```
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Avoid unused imports such as 'java.io.PrintWriter'](https://www.codacy.com/app/SeLion/paypal-SeLion/file/2044430110/issues/source?bid=2928425&fileBranchId=3110989#l31)

sundaramrajendran committed:
Additonal improvements for the existing servlets

- All of our console pages html content loaded
  from templates instead of html code
  intermixed in the java code
- All of our html, js, css and image resources
  moved to new path under
  src/main/resources/com/paypal/selion
- Added SeLion Artwork logo to all console pages
- Updated GridForceRestartServlet as inject
  application/json into the template which jquery
  javascript uses to render the page
- Refactoring efforts around ServletHelper methods
- Unrelated codacy reported issues - clean up

mach6 commented on server/src/main/java/com/paypal/selion/grid/servlets/GridAutoUpgradeDelegateServlet.java:
```diff
@@ -160,24 +160,12 @@ protected void process(HttpServletRequest request, HttpServletResponse response)
-                writer.write("</form>");
-                writer.write("</body>");
-                writer.write("</html>");
+                ServletHelper.respondWithTemplate(writer, PENDING_NODE_FILE, ids, ids, downloadJSON);
```
I'd still like this to be further refactored. Rather than passing in the `writer`, I'd like us to pass the `response` object and all the input for consideration when writing the response. I'll go ahead and do the changes to this branch.

mach6 commented on server/src/main/java/com/paypal/selion/utils/ServletHelper.java:
```diff
@@ -115,124 +118,26 @@ public static void respondAsHtmlUsingJsonAndTemplateWithHttpStatus(HttpServletRe
-        writer.write("<img id='top' src='/grid/resources/form/top.png' alt='' >");
+    public static void respondWithTemplate(PrintWriter writer, String page, Object... args)
+            throws IOException {
+        String template = IOUtils.toString(LoginServlet.class.getResourceAsStream(page), "UTF-8");
```
`LoginServlet.class` should not be used here. I will fix this and update the branch

mach6 commented:
@sundaramrajendran FYI this looks good to me. I'm currently working on a branch to test these changes both pre commit and post commit. The branch is in my repo here -- https://github.com/mach6/SeLion/tree/mach6-servlet-tests ... once this work is complete (and reviewed), we can sqaush/merge this PR

sundaramrajendran commented:
@mach6 Thanks

Commit:
Update PasswordChangeServlet, fix #248, servlet tests, and copyrights.

- PasswordChangeServlet now requires a logged in session
- Servlet tests updated to use validateHtmlResponseContent instead of
  the deprecated validateResultingPage method -- since servlets now
  correctly set a content-type and character encoding. Removed the
  deprecated method validateResultingPage from
  BaseGridRegistryServletTest
- Fixed #248 making look/feel more consistent
- Update copyrights to include 2016 for files touched

Commit:
Update the Readme.md to include example for sauce proxy boot

mach6 commented on server/src/main/java/com/paypal/selion/grid/AppiumSpawner.java:
```diff
@@ -41,19 +41,19 @@ public AppiumSpawner(String[] args, ProcessLauncherOptions options) {
     }
 
-    public static final void main(String[] args) {
+    public static void main(String[] args) {
```
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Consider using varargs for methods or constructors which take an array the last parameter.](https://www.codacy.com/app/SeLion/paypal-SeLion/file/2103039166/issues/source?bid=2928425&fileBranchId=3110989#l44)

mach6 replied:
will not fix

mach6 replied:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Consider using varargs for methods or constructors which take an array the last parameter.](https://www.codacy.com/app/SeLion/paypal-SeLion/file/2103142558/issues/source?bid=2928425&fileBranchId=3110989#l44)

mach6 commented on server/src/main/java/com/paypal/selion/grid/IOSDriverJarSpawner.java:
```diff
@@ -45,12 +45,12 @@ public IOSDriverJarSpawner(String[] args, ProcessLauncherOptions options) {
     }
 
-    public static final void main(String[] args) {
+    public static void main(String[] args) {
```
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Consider using varargs for methods or constructors which take an array the last parameter.](https://www.codacy.com/app/SeLion/paypal-SeLion/file/2103039170/issues/source?bid=2928425&fileBranchId=3110989#l48)

mach6 replied:
will not fix

mach6 replied:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Consider using varargs for methods or constructors which take an array the last parameter.](https://www.codacy.com/app/SeLion/paypal-SeLion/file/2103142562/issues/source?bid=2928425&fileBranchId=3110989#l48)
